src/lib/db/context-field-store.ts

```diff
@@ -88,15 +98,17 @@ class ContextFieldStore implements IContextFieldStore {
         return present;
     }
 
+    // TODO: write tests for the changes you made here?
     async create(contextField: IContextFieldDto): Promise<IContextField> {
-        const row = await this.db(TABLE)
+        const [row] = await this.db(TABLE)
             .insert(this.fieldToRow(contextField))
             .returning('*');
+
         return mapRow(row);
     }
 
     async update(data: IContextFieldDto): Promise<IContextField> {
-        const row = await this.db(TABLE)
+        const [row] = await this.db(TABLE)
```

Comment:
As far as I understand, neither of these worked correctly before. That is, `mapRow` on a list of rows wouldn't work. However, I don't think we ever used the API to get this data back. 

There doesn't seem to be any tests concerned with this, but I would be more than happy to add them. However, I'm not sure exactly where and what would be a good way to test them? Input is very welcome.

Reply:
Yeah, I'd definitely sleep better at night if we had a test to poke this with a stick. I want to say that you're right but without a strong, static type system it's an educated guess.

Also don't see any tests but the contents of src/test/e2e/stores/ is probably a good model to build some tests off of. I don't think this needs a massively complex test, just a write, update and then a read back to confirm that what you expect has happened. Happy to help if that's not enough info, grab me on Slack

Reply:
Sweet! I'll have a look and see what I can do. Just to be clear: the manual testing I did (which uncovered that it didn't work) was that I tried to use the data that the method returned. Turns out it was all undefined except for `legalValues: []`. So this fixes the mapping, but yeah, I'll look into the testing bit 👷🏼

Reply:
Addressed in 420546ac759abb48c98587661feba6cb800c6077